src/components/ConnectionsListProfiles.tsx

```diff
@@ -15,18 +20,32 @@ enum ListStatus {
 }
 
 interface ConnectionsListProfilesProps {
+  userId: DSNPUserId;
   listStatus: ListStatus;
-  following: Record<DSNPUserId, boolean>;
-  followers: Record<DSNPUserId, boolean>;
+  following: Record<DSNPUserId, RelationshipStatus>;
+  followers: Record<DSNPUserId, RelationshipStatus>;
 }
 
 const ConnectionsListProfiles = ({
+  userId,
   listStatus,
   following,
   followers,
 }: ConnectionsListProfilesProps): JSX.Element => {
-  const userRelationship = (userProfile: Profile) =>
-    following[userProfile.fromId] ? "Unfollow" : "Follow";
+  const dispatch = useAppDispatch();
+
+  const isFollowing = (userProfile: Profile): boolean =>
```

Comment:
There are a few things below that harp on this, but `userProfile` makes me thing it is the logged in user or the user of the current profile, not the connection user which is what I think it is.